### DISCUSSION TITLE
Jamulii - a Jamulus Server service

### DISCUSSION THREAD

**Discussion by ginjo** [jamulussoftware/jamulus#3205] [General]
Hello Jamulus community,

Jamulus has been a valuable tool in my musical life, allowing my friends, my band, and myself to make music in ways that would not have been possible without it. So a big thanks to the devs for creating and maintaining Jamulus!

To help streamline part of the technical equation, I developed a core set of tools to manage our Jamulus Servers. Over the last couple years, I've expanded this project into the cloud-based [Jamulii service](https://jamulii.com), allowing anyone to run Jamulus Server *instances* for public and private sessions. Additional capabilities include automatic recording mixdown, network optimization, and download links. Scheduled servers, social collaboration, cloud storage (dropbox, google drive, etc.), mixdown effects, and more are on the to-do list.

I'd love for people to get their hands on it, and give it a spin. The [jamulii.com](https://jamulii.com) service is in beta mode, so payment setup is not currently required (it's free). Servers are available in North America and Western Europe, with more locations planned around the globe.

You can [read more](https://jamulii.com/jamulii/about) about it, and feel free to leave [feedback](https://jamulii.com/jamulii/contact).

**Comment by ann0see** [comment]
Hi, sounds interesting. We already have Koord and Melomax which provide server hosting, but having multiple is good. 

> All server instances have a maximum up-time of 240 minutes. After that time, server instances will be stopped and reset.

I could see that as a problem for some people. 

I'd maybe include screenshots on https://jamulii.com/jamulii/usage showing the UI.

**Reply by ginjo** [reply]
Hi, thanks for the tips! If you don't mind me asking, what aspect of the time limit do you see as being an issue for some? Length? Mere existence of? I'm not set on any particular time limit, but during testing I would like to prevent long-lived servers that are just sitting idle. In production, there will be no up-time limit.

The initial rollout is targeted more toward on-demand users who need a server for a specific session or event. But I'd love to host long-running servers for users who want always-up availability. That's mostly just a matter of creating a pricing plan to fit that use case.

Good point about the visuals. I'll be putting something together for that.

**Reply by mcfnord** [reply]
Maybe there can be a 6 hour limit, but a script can close it earlier if it sees everyone has left.

**Reply by ginjo** [reply]
You read my mind 😉 ,  I increased the timeout (for contiguous up-time, during the beta phase) to 6 hours. Users are welcome to start up a new instance if they want more time.

Server instances currently have an idle timeout of 10 minutes, after which the server is stopped. Idle is defined as < 1% CPU usage (which is pretty consistent with no Jamulus clients connected). Stopped instances can be restarted in the UI. If a stopped instance is not restarted within 5 minutes, it will be removed/deleted (all recording data is saved, regardless of the instance lifecycle). The idle-timeout will eventually be user-adjustable.

**Reply by ann0see** [reply]
That's great! The time limit should not be noticeable by people and in my opinion be - if implemented- just to prevent abuse.

**Reply by ginjo** [reply]
Agreed! I would never want to cut short someone's active session.

And just to reiterate, for those who may have missed it above, there will be no limit on uptime, once the service is launched in production.
